src/ServiceProvider.php

```diff
@@ -43,7 +45,11 @@ public function register()
                 'credentials' => new Adapter($app->make(CacheManager::class), config('easyaws.cache_store')),
                 'client' => config('easyaws.http_client'), // NOTE: used for unit testing only
             ];
-            return CredentialProvider::defaultProvider($credentialsCache);
+
+            return CredentialProvider::chain(
```

Comment:
It might be useful to wrap this in memoize as well, just like the defaultProvider. Although, the performance increase is probably negligible since if it does use the getLaravelEnvProvider method to resolve credentials, memoizing those doesn't do much. Let me know what you think